File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/Event.java
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReadableMap;

/**
 * Event interface.
 */
public interface Event {

    /**
     * The event name.
     *
     * @return The event name.
     */
    @NonNull
    String getName();

    /**
     * The event body.
     *
     * @return The event body.
     */
    @NonNull
    ReadableMap getBody();

    /**
     * Flags if the event should only be delivered in the foreground or not.
     *
     * @return {@code true} if the event should only be delivered in the foreground, otherwise {@code false}.
     */
    boolean isForeground();
}

File: urbanairship-react-native/android/src/main/java/com/urbanairship/reactnative/Event.kt
/* Copyright Urban Airship and Contributors */

package com.urbanairship.reactnative

import com.facebook.react.bridge.ReadableMap

/**
 * Event interface.
 */
interface Event {

    /**
     * The event name.
     */
    val name: String

    /**
     * The event body.
     */
    val body: ReadableMap

    /**
     * Flags if the event should only be delivered in the foreground or not.
     */
    val isForeground: Boolean
}